From bdfd063e797efbd9371ef284cb1178865ccdaf63 Mon Sep 17 00:00:00 2001
From: HolyWu <holywu@gmail.com>
Date: Mon, 17 May 2021 11:08:55 +0800
Subject: [PATCH 3/4] Revert "avcodec/h264_parser: set missing pts for
 top/bottom field frames"

This reverts commit 01fa4fb69e40165c1e03bcf6939e7f4ad07b69b6.

Somehow the made-up timestamp caused seeking issues in H.264 PAFF.
---
 libavcodec/h264_parser.c | 22 ----------------------
 1 file changed, 22 deletions(-)

diff --git a/libavcodec/h264_parser.c b/libavcodec/h264_parser.c
index aacd44cf3b..285177e8f7 100644
--- a/libavcodec/h264_parser.c
+++ b/libavcodec/h264_parser.c
@@ -60,7 +60,6 @@ typedef struct H264ParseContext {
     uint8_t parse_history[6];
     int parse_history_count;
     int parse_last_mb;
-    int64_t reference_dts;
     int last_frame_num, last_picture_structure;
 } H264ParseContext;
 
@@ -619,26 +618,6 @@ static int h264_parse(AVCodecParserContext *s,
         s->flags &= PARSER_FLAG_COMPLETE_FRAMES;
     }
 
-    if (s->dts_sync_point >= 0) {
-        int64_t den = avctx->time_base.den * (int64_t)avctx->pkt_timebase.num;
-        if (den > 0) {
-            int64_t num = avctx->time_base.num * (int64_t)avctx->pkt_timebase.den;
-            if (s->dts != AV_NOPTS_VALUE) {
-                // got DTS from the stream, update reference timestamp
-                p->reference_dts = s->dts - av_rescale(s->dts_ref_dts_delta, num, den);
-            } else if (p->reference_dts != AV_NOPTS_VALUE) {
-                // compute DTS based on reference timestamp
-                s->dts = p->reference_dts + av_rescale(s->dts_ref_dts_delta, num, den);
-            }
-
-            if (p->reference_dts != AV_NOPTS_VALUE && s->pts == AV_NOPTS_VALUE)
-                s->pts = s->dts + av_rescale(s->pts_dts_delta, num, den);
-
-            if (s->dts_sync_point > 0)
-                p->reference_dts = s->dts; // new reference
-        }
-    }
-
     *poutbuf      = buf;
     *poutbuf_size = buf_size;
     return next;
@@ -696,7 +675,6 @@ static av_cold int init(AVCodecParserContext *s)
 {
     H264ParseContext *p = s->priv_data;
 
-    p->reference_dts = AV_NOPTS_VALUE;
     p->last_frame_num = INT_MAX;
     ff_h264dsp_init(&p->h264dsp, 8, 1);
     return 0;
-- 
2.30.1.windows.1

